geolocator_android/android/src/main/java/com/baseflow/geolocator/location/LocationManagerClient.java

```diff
@@ -73,45 +75,35 @@ static boolean isBetterLocation(Location location, Location bestLocation) {
     return false;
   }
 
-  private static String getBestProvider(
-      LocationManager locationManager, LocationAccuracy accuracy) {
-    Criteria criteria = new Criteria();
-
-    criteria.setBearingRequired(false);
-    criteria.setAltitudeRequired(false);
-    criteria.setSpeedRequired(false);
-
-    switch (accuracy) {
-      case lowest:
-        criteria.setAccuracy(Criteria.NO_REQUIREMENT);
-        criteria.setHorizontalAccuracy(Criteria.NO_REQUIREMENT);
-        criteria.setPowerRequirement(Criteria.NO_REQUIREMENT);
-        break;
-      case low:
-        criteria.setAccuracy(Criteria.ACCURACY_COARSE);
-        criteria.setHorizontalAccuracy(Criteria.ACCURACY_LOW);
-        criteria.setPowerRequirement(Criteria.NO_REQUIREMENT);
-        break;
-      case medium:
-        criteria.setAccuracy(Criteria.ACCURACY_COARSE);
-        criteria.setHorizontalAccuracy(Criteria.ACCURACY_MEDIUM);
-        criteria.setPowerRequirement(Criteria.POWER_MEDIUM);
-        break;
-      default:
-        criteria.setAccuracy(Criteria.ACCURACY_FINE);
-        criteria.setHorizontalAccuracy(Criteria.ACCURACY_HIGH);
-        criteria.setPowerRequirement(Criteria.POWER_HIGH);
-        break;
-    }
-
-    String provider = locationManager.getBestProvider(criteria, true);
-
-    if (provider.trim().isEmpty()) {
-      List<String> providers = locationManager.getProviders(true);
-      if (providers.size() > 0) provider = providers.get(0);
-    }
+  private static @Nullable String getBestProvider(@NonNull LocationManager locationManager) {
+      final List<String> enabledProviders = locationManager.getProviders(true);
+
+      if (enabledProviders.contains(LocationManager.FUSED_PROVIDER) && Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+          return LocationManager.FUSED_PROVIDER;
+      } else if (enabledProviders.contains(LocationManager.GPS_PROVIDER)) {
+          return LocationManager.GPS_PROVIDER;
+      } else if (enabledProviders.contains(LocationManager.NETWORK_PROVIDER)) {
+          return LocationManager.NETWORK_PROVIDER;
+      } else if (!enabledProviders.isEmpty()){
+          return enabledProviders.get(0);
+      } else {
+          return null;
+      }
+  }
 
-    return provider;
+  private static @LocationRequestCompat.Quality int accuracyToQuality(@NonNull LocationAccuracy accuracy) {
+      switch (accuracy) {
+          case lowest:
```

Comment:
I think `lowest` should return `LocationRequestCompat.PASSIVE_INTERVAL`:

```suggestion
          case lowest:
                  return LocationRequestCompat.PASSIVE_INTERVAL;
```

Reply:
Why do you think this is not a quality parameter? According to the [documentation](https://developer.android.com/reference/androidx/core/location/LocationRequestCompat#PASSIVE_INTERVAL()) the `LocationRequestCompat.PASSIVE_PROVIDER` is a valid quality parameter with the const value  of `9223372036854775807 `.

Reply:
You are correct, I miss read the documentation. This should be good the way it is.